ProcessLib/LiquidFlow/LiquidFlowLocalAssembler-imp.h

```diff
@@ -0,0 +1,136 @@
+
+#include "NumLib/Function/Interpolation.h"
+#include "MaterialLib/PhysicalConstant.h"
+
+namespace ProcessLib
+{
+namespace LiquidFlow
+{
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned GlobalDim>
+void LiquidFlowLocalAssembler<ShapeFunction, IntegrationMethod, GlobalDim>::
+    assemble(double const /*t*/, std::vector<double> const& local_p,
+             std::vector<double>& local_M_data,
+             std::vector<double>& local_K_data,
+             std::vector<double>& local_b_data)
+{
+    auto const local_matrix_size = local_p.size();
+    assert(local_matrix_size == ShapeFunction::NPOINTS);
+
+    auto local_M = MathLib::createZeroedMatrix<NodalMatrixType>(
+        local_M_data, local_matrix_size, local_matrix_size);
+    auto local_K = MathLib::createZeroedMatrix<NodalMatrixType>(
+        local_K_data, local_matrix_size, local_matrix_size);
+    auto local_b = MathLib::createZeroedVector<NodalVectorType>(
+        local_b_data, local_matrix_size);
+
+    // For velocity calculation:
+    // TODO: move velocity calculation to a member function, and call the
+    // function after the linearization step if the calculation is needed.
+    const auto local_p_vec =
+        MathLib::toVector<NodalVectorType>(local_p, local_matrix_size);
+
+    unsigned const n_integration_points =
+        _integration_method.getNumberOfPoints();
+
+    const unsigned mat_id = 0; // TODO for heterogeneous medium
+    const Eigen::MatrixXd& perm =
+        _material_properties.intrinsic_permeabiliy[mat_id];
+
+    // TODO: The following two variables should be calculated inside the
+    //       the integration loop for non-constant porosity and storage models.
+    double var4porosity = 0.;
+    double var4storage = 0.;
+    for (unsigned ip = 0; ip < n_integration_points; ip++)
+    {
+        auto const& sm = _shape_matrices[ip];
+        auto const& wp = _integration_method.getWeightedPoint(ip);
+
+        double p = 0.;
+        NumLib::shapeFunctionInterpolate(local_p, sm.N, p);
+        // TODO : compute _temperature from the heat transport pcs
+
+        const double integration_factor
+                               = sm.integralMeasure * sm.detJ * wp.getWeight();
+
+        // Assemble mass matrix, M
+        local_M.noalias() +=
+            _material_properties.getMassCoefficient(var4porosity, var4storage,
+                                                   p, _temperature, mat_id) *
+                                 sm.N.transpose() * sm.N * integration_factor;
+
+        // Compute density:
+        const double rho_g =
+            _material_properties.getLiquidDensity(p, _temperature) *
+            MaterialLib::PhysicalConstant::g;
+        // Compute viscosity:
+        const double mu = _material_properties.getViscosity(p, _temperature);
+
+        // Assemble Laplacian, K, and RHS by the gravitational term
+        if (perm.size() == 1)  // Save the computing time for isotropic permeability.
```

Comment:
I think for 3D elements, such as hexehadra, it could save the computing time for a homogeneous medium. Image to compare the commutating times of 
- Matrix with a size of (20, 3) multiplied with  a matrix with a size of (3, 3)
- Matrix with a size of (20, 3) multiplied with  a scalar number.

To make this function concise, the computation of velocity, which will be in the next PR, is removed.